api/admin/controller/__init__.py

```diff
@@ -2510,18 +2511,46 @@ def _is_email(self, email):
     def validate_url(self, settings):
         """Find any URLs that the user has submitted, and make sure that
         they are in a valid format."""
-        settings = settings or self._get_settings()
-        if isinstance(settings, (list,)):
-            # Find the fields that have to do with URLs
-            url_fields = filter(lambda s: (s.get("format") == "url" or s.get("key") == "url") and flask.request.form.get(s.get("key")), settings)
-            # Narrow the URL-related fields down to the ones for which the user actually entered a value
-            url_inputs = [flask.request.form.get(field.get("key")) for field in url_fields]
-        else:
-            # The SitewideSettingsController passes in a string
-            url_inputs = [settings]
+        # Find the fields that have to do with URLs and are not blank.
+        # If this is a sitewide setting, then the user input needs to be accessed
+        # via "value" rather than via the setting's key.
+        url_fields = filter(lambda s: s.get("format") == "url" and
+                            (flask.request.form.get(s.get("key")) or flask.request.form.get("value"))
+                            , settings)
+        # Narrow the URL-related fields down to the ones for which the user actually entered a value
+        url_inputs = [(flask.request.form.get(field.get("key")) or flask.request.form.get("value")) for field in url_fields]
+
         for url in url_inputs:
             if not self._is_url(url):
                 return INVALID_URL.detailed(_('"%(url)s" is not a valid URL.', url=url))
 
     def _is_url(self, url):
         return any([url.startswith(protocol + "://") for protocol in "http", "https"])
+
+    def validate_number(self, settings):
+        """Find any numbers that the user has submitted, and make sure that they are 1) actually numbers,
+        2) positive, and 3) lower than the specified maximum, if there is one."""
+        # Find the fields that should have numeric input and are not blank.
+        number_fields = filter(
+                            lambda s: (s.get("type") == "number") and
+                            (flask.request.form.get(s.get("key")) or flask.request.form.get("value"))
```

Comment:
I'm not crazy about putting in these special cases for the sitewide settings controller, where the value of the setting is always in `"value"` instead of in a variable taken from the setting's `key`. But I don't have a better idea that's easy to implement. Maybe @aslagle has an idea.

Reply:
I guess you could change the front-end code for sitewide settings to send the value in a parameter with the key instead. Maybe that code could be made more consistent with the other types of settings.

Reply:
I agree that it should be consistent with the other types of settings; I'll make a separate branch to look into that!

Reply:
@aslagle Was there a particular rationale for the front-end sitewide settings code not being consistent with the other settings in the first place (i.e. is there something that I should be particularly careful not to break by messing around with this!)?

Reply:
Not really, it was just because those settings aren't associated with an integration or a library so there's always a single field in the form. As long as you're careful to test after you change things it should be fine.